Frameworks/UIKit/UIImage.mm

```diff
@@ -44,487 +44,416 @@ @interface UIImage ()
     }
 
     EbrFread(in, 1, 8, fpIn);
 
     EbrFclose(fpIn);
 
-    if (in[0] == 0x89 && in[1] == 'P' && in[2] == 'N' && in[3] == 'G') {
-        m_pImage = CGPNGImageCreateFromFile([NSString stringWithCString:pathStr]);
-    } else if ((in[0] == 0xFF && in[1] == 0xD8) || (in[0] == 0xD8 && in[1] == 0xFF)) {
-        m_pImage = CGJPEGImageCreateFromFile([NSString stringWithCString:pathStr]);
-        if (((CGJPEGImageBacking*)m_pImage->Backing()) && ((CGJPEGImageBacking*)m_pImage->Backing())->_orientation) {
-            [self setOrientation:((CGJPEGImageBacking*)m_pImage->Backing())->_orientation];
-        }
-    } else {
-        EbrFile* fpIn;
+    fpIn = EbrFopen(pathStr, "rb");
+    if (!fpIn) {
+      TraceVerbose(TAG, L"Image %hs not found", pathStr);
+      [self release];
+      return nil;
+    }
 
-        fpIn = EbrFopen(pathStr, "rb");
-        if (!fpIn) {
-            TraceVerbose(TAG, L"Image %hs not found", pathStr);
-            // m_pImage = new CGBitmapImage(64, 64, __CGSurfaceFormat::_ColorABGR, NULL);
-            return nil;
-        }
+    EbrFseek(fpIn, 0, SEEK_END);
+    int len = EbrFtell(fpIn);
+    if (len <= 0) {
+      TraceVerbose(TAG, L"Image %hs invalid", pathStr);
+      EbrFclose(fpIn);
 
-        EbrFseek(fpIn, 0, SEEK_END);
-        int len = EbrFtell(fpIn);
-        if (len <= 0) {
-            TraceVerbose(TAG, L"Image %hs invalid", pathStr);
-            // m_pImage = new CGBitmapImage(64, 64, __CGSurfaceFormat::_ColorABGR, NULL);
-            EbrFclose(fpIn);
-            return nil;
-        }
-        EbrFseek(fpIn, 0, SEEK_SET);
-
-        auto inBuffer = std::make_unique<char[]>(len);
-        len = EbrFread(inBuffer.get(), 1, len, fpIn);
-
-        EbrFclose(fpIn);
-
-        if (!loadTIFF(self, inBuffer.get(), len)) {
-            if (!loadGIF(self, inBuffer.get(), len)) {
-                if (!loadBMP(self, inBuffer.get(), len)) {
-                    TraceVerbose(TAG, L"Unrecognized image");
-                    for (int i = 0; i < MIN(len, 64); i++) {
-                        TraceVerbose(TAG, L"%02x ", inBuffer[i]);
-                        if ((i + 1) % 16 == 0)
-                            TraceVerbose(TAG, L"");
-                    }
-                    TraceVerbose(TAG, L"Image type %hs not recognized header=%x", pathStr, *((DWORD*)in));
-                    // m_pImage = new CGBitmapImage(64, 64, __CGSurfaceFormat::_ColorABGR, NULL);
-                    return nil;
-                }
-            }
-        }
+      [self release];
+      return nil;
+    }
+    EbrFseek(fpIn, 0, SEEK_SET);
+
+    auto inBuffer = std::make_unique<char[]>(len);
+    len = EbrFread(inBuffer.get(), 1, len, fpIn);
+
+    EbrFclose(fpIn);
+
+    m_pImage = _CGImageGetImageFromData(inBuffer.get(), len);
+
+    if (!m_pImage) {
+      TraceVerbose(TAG, L"Unrecognized image");
+      for (int i = 0; i < MIN(len, 64); i++) {
+        TraceVerbose(TAG, L"%02x ", inBuffer[i]);
+        if ((i + 1) % 16 == 0)
```

Comment:
debug verbose logs from before (incase the UI team is depending on this).
not the context of this CR.

---
In reply to: [84146123](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84146123) [](ancestors = 84146123)